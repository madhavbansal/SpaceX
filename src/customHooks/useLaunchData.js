import React, { useState, useEffect } from 'react';
import axios from 'axios';
import { useLocation } from "react-router-dom";
import queryString from 'query-string';
import { constructApiUrl, baseUrl } from '../utils/helper';


const useLaunchData = () => {
    const location = useLocation();
    let params = queryString.parse(location.search, { parseBooleans: true });
    const newUrl = constructApiUrl(baseUrl, params);

    
    const [options, setOptions] = useState({ loading: true, error: false, data: [] });


    useEffect(() => {
        axios.get(newUrl).then((response) => {
            const { data = [] } = response || {};
            setOptions({ loading: false, error: false, data });
        })
            .catch(error => {
                console.log(error);
                setOptions({ loading: false, error: true, data: [] });
            });
    }, [newUrl]);
    return options;
}

export default useLaunchData;
