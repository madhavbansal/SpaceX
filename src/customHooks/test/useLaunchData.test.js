import React from 'react';
import renderer from 'react-test-renderer';
import useLaunchData from '../useLaunchData';


jest.mock('react-router-dom', () => {
  return {
    useLocation: () => ({}),
    useHistory: () => ({})
  }
})

jest.mock('query-string', () => {
  return {
    parse:()=>({})
  }
})

test('Test the useLaunchData Component', () => {
  const component = renderer.create(
    <useLaunchData />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});