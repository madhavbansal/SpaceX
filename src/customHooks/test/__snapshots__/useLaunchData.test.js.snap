// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the useLaunchData Component 1`] = `<useLaunchData />`;
